pkg/controller/account/account_controller.go

```diff
@@ -148,14 +148,11 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	// We expect this secret to exist in the same namespace Account CR's are created
-	awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+	// We build the input here but the client later because we handle the error differently
```

Comment:
I think that this comment is misleading. My first thought was, "we handle the errors exactly the same...", but we do build the client in two places so IMO I would just leave it like:
```
// Defining the AWS client builder input
```